src/API/index.js

```diff
@@ -11,13 +11,13 @@ export const fetchPubkeysList = async () => {
   return data;
 };
 
-export const fetchInstanceTypesList = async (region) => {
-  const { data } = await axios.get(provisioningUrl(`instance_types/aws?region=${region}`));
+export const fetchInstanceTypesList = async (region, provider) => {
+  const { data } = await axios.get(provisioningUrl(`instance_types/${provider}?region=${region}`));
   return data;
 };
 
-export const createAWSReservation = async (params) => {
-  return axios.post(provisioningUrl('reservations/aws'), params);
+export const createReservation = (provider) => async (params) => {
```

Comment:
could we keep the functions here consistent? I know this is bit better, but in other cases we need parameters, we do `() => createReservation(provider)` on the caller (in the `useQuery`).